### PR TITLE
Localized all messages and the argument types.

### PR DESCRIPTION
**Pull request by jstuyts**
I made the following changes to how things worked up to now:

* The message from the exception thrown when a type cannot be converted, is no longer appended to the error message. The exception messages are technical and not localized, so they were unlikely to help end users. I looked at `getLocalizedMessage()`, but this also did not provide a localized message (in Dutch), and that method would use the default locale of the JVM instead of the locale of the parser. See the changes in  `ConstructorArgumentType` and `ReflectArgumentType`.
* I introduced friendlier names for the primitive types as the end user may not be familiar with the meanings of the Java names.

### CODE REVIEW

File: src/main/java/net/sourceforge/argparse4j/inf/ArgumentParserException.java [hunk 1]
```diff
@@ -23,6 +23,9 @@
  */
 package net.sourceforge.argparse4j.inf;
 
+import static net.sourceforge.argparse4j.internal.MessageLocalization.localizeIfPossible;
```
Comment (tatsuhiro-t) [inline src/main/java/net/sourceforge/argparse4j/inf/ArgumentParserException.java]: In my opinion, inf package should not depend on internal package.  I mean that concrete package depends on abstract package, but not the other way around.  But there are already some cycles here and there.  So I'll do some experiments moving classes around packages to clean up dependencies after merging this PR.

Perhaps, it is my mistake to put ArgumentParserException class in inf package.  It should have been interface.

File: src/main/java/net/sourceforge/argparse4j/internal/TypeNameLocalization.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.ConfiguredArgumentParser;
+
+/**
+ * <p>
+ * Localization of names for argument types. Localized names for commonly used
+ * ypes in the JDK (e.g. {@code Integer} and {@code Boolean}) are provided, but
```
Comment (tatsuhiro-t) [inline src/main/java/net/sourceforge/argparse4j/internal/TypeNameLocalization.java]: I suppose it is 'types' not 'ypes'

Comment (jstuyts) [inline src/main/java/net/sourceforge/argparse4j/internal/TypeNameLocalization.java]: Yes, it must be _types_.

File: src/main/java/net/sourceforge/argparse4j/inf/ArgumentParserException.java (hunk 1 above)
Comment (jstuyts) [inline src/main/java/net/sourceforge/argparse4j/inf/ArgumentParserException.java]: Let me know if I can help.

Personally I am not too bothered by cyclic dependencies within in the same module. `internal` contains the implementation code that is shared by types in multiple public packages. You do not want to make package private copies of that implementation code in each public package, so you need an additional package to put the code in. Logically this implementation package will need to refer to the types in the public packages too, so cycles are sort of inevitable.

I am sure that it is possible to create a package structure where cycles do not occur, but in my opinion the advantages of such as structure do not outweigh the possibly strange code constructs needed to achieve it.

Comment (tatsuhiro-t) [inline src/main/java/net/sourceforge/argparse4j/inf/ArgumentParserException.java]: Thank you.
This is just my personal preference, and it is not the end of the world with dependency cycles.
It looks like by moving a couple of classes, I can achieve this in argparse4j.
If it requires substantial changes, I'll give up.

File: src/main/java/net/sourceforge/argparse4j/internal/TypeNameLocalization.java (hunk 2 above)
Comment (tatsuhiro-t) [inline src/main/java/net/sourceforge/argparse4j/internal/TypeNameLocalization.java]: If you are OK, I just fixup this, and merge this PR.

Comment (jstuyts) [inline src/main/java/net/sourceforge/argparse4j/internal/TypeNameLocalization.java]: Yes, fine with me. Thanks again.

**Comment by tatsuhiro-t** [comment]
While looking at the code, I wonder that it might be simpler to add getConfig() to ArgumentParser interface without introducing ConfiguredArgumentParser.
I'm not familiar with Java ABI, but I've added several new methods to public interface in the course of the development of this library.
Since the only producer of ArgumentParser implementation is by ArgumentParsers, it could be safe to do that.

Again, this is still my personal preference, so I will see it could help code base after merging PR.

Thank you.

**Comment by jstuyts** [comment]
Added `getConfig()` to the existing interface would be a lot simpler. But I went for full backward compatibility. Adding a method to an interface will prevent compilation of external implementations of `ArgumentParser`, and will result in `NoSuchMethodError` when the new method is called on a class that does not implement it.

If more flexibility is needed, what is stable and what is not should be defined. That way you can say things like: You can call this interface, but extending/implementing it is not supported as we want to be able to add methods to it.

**Comment by tatsuhiro-t** [comment]
I agree that that should be defined.
What I have in my mind is that adding interface is OK-ish.  Without that we have to invent new interface when every time new method is added.

Anyway, thank you for PR.  Now it is merged.

**Pull request merged**